plugins/readme/src/api/ReadmeClient.tsx

```diff
@@ -1,10 +1,45 @@
 import {
+  ApiHolder,
   DiscoveryApi,
   FetchApi,
   IdentityApi,
 } from '@backstage/core-plugin-api';
-import { DEFAULT_NAMESPACE, parseEntityRef } from '@backstage/catalog-model';
-import { ReadmeApi } from './ReadmeApi';
+import {
+  DEFAULT_NAMESPACE,
+  Entity,
+  parseEntityRef,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import { NotFoundError } from '@backstage/errors';
+import { ReadmeApi, readmeApiRef } from './ReadmeApi';
+
+/**
+ * Checks if a README is available for the given entity by making a request to the backend.
+If the backend returns a 404 NotFound error, it indicates that no README is available for the entity.
+@param entity - The entity for which to check the README availability.
+@param context - The context providing access to the API holder.
+@returns A promise that resolves to true if the README is available, or false if not found (404).
+ *  @public
+ */
+export const isReadmeAvailable = async (
+  entity: Entity,
+  context: { apis: ApiHolder },
+): Promise<boolean> => {
+  const readmeClient = context.apis.get(readmeApiRef);
+
+  if (readmeClient === undefined) {
+    return false;
+  }
+
+  try {
+    await readmeClient.getReadmeContent(stringifyEntityRef(entity));
+  } catch (error) {
+    if (error instanceof NotFoundError) {
```

Comment:
Don't the backend need to throw a "NotFoundError" for this to work? The readme backend should use the proper backstage errors.

Reply:
I think upstream uses this pattern to check the error type?
```typescript
if (isError(error) && error.name === 'NotFoundError') {
```

Reply:
Something like that? I've pushed a few new commits.